miio/integrations/vacuum/roborock/vacuum.py

```diff
@@ -403,7 +403,10 @@ def manual_control(
     @command()
     def status(self) -> VacuumStatus:
         """Return status of the vacuum."""
-        return VacuumStatus(self.send("get_status")[0])
```

Comment:
I will try to test this tonight

Reply:
I'm slowly working the backlog after returning back home. You have a good point about having separate functions to make it possible to avoid unnecessary I/O that I was not considering while working on the initial implementation. This is not only relevant to save some I/O requests, but the current way has another issue, namely that handling of error cases (e.g., a device not supporting some feature) needs to be done case-by-case...

Instead of embedding, another approach could be to extend the API to add a way to expose references to methods that can be used to obtain the status information, and leave it to the downstream like homeassistant to choose if they want to use it. The problem I'm seeing though is that when the sensors & co are dynamic, homeassistant would need to hardcode what it wants as long as there is no way to check which entities are disabled, so I don't think that's a good idea for now. Your solution to add a separate "raw update" for cli and other users to allow obtaining just the `get_status` results sounds good to me.

Back to the issue of error handling / avoiding unsupported calls: maybe the `embed` should rather acts as a builder and take a callable instead of a status container as an input. This would make it easier to implement generic handling to avoid crashing on unsupported features, as well as to avoid doing unnecessary I/O on subsequent update cycles. I think this needs some thinking and experimenting to see what works the best.